helix/appendix/alternate-visual-studio-structures.rst

```diff
@@ -0,0 +1,231 @@
+be the case for simple modules which only contain configuration and/or View renderings.
+These add no value and unecessarily slow down your builds.
+
+The same applies for TDS projects with no items. If these exist only to build/deploy a module,
+consider a single build-only TDS project with `multiple Source Web Projects <http://hedgehogdevelopment.github.io/tds/chapter4.html#general>`__.
+
+.. admonition:: Sitecore Helix Examples
+
+    The TDS version of Helix Basic Company uses a single TDS project for
+    file copy / packaging, and additional TDS projects as-needed within each module.
+    It also only has test projects for modules with testable logic.
+
+    .. figure:: _static/basic-company-projects.png
+        :scale: 75%
+
+Are there inefficiencies in your build scripts?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Previous versions of Sitecore Habitat included gulp scripts which built each
+Visual Studio project individually, rather than the entire solution at once.
+This could lead to much slower builds as project dependencies could be built
+more than once. Ensure your build scripts target the VS Solution or a single
+"root-level" project.
+
+.. admonition:: Sitecore Habitat
+
+    Newer versions of `Sitecore Habitat <https://github.com/Sitecore/Habitat>`__
+    have an updated ``gulpfile`` which targets the entirety of the Visual Studio
+    Solution for builds.
+
+    .. code-block:: js
+
+        gulp.task("Publish-All-Projects",
+            function(callback) {
+                var msbuild = new _msbuild(callback);
+                msbuild.solutionName = config.solutionName + '.sln';
+                var overrideParams = [];
+                overrideParams.push('/p:Configuration=' + config.buildConfiguration);
+                overrideParams.push('/p:DeployOnBuild=true');
+                overrideParams.push('/p:DeployDefaultTarget=WebPublish');
+                overrideParams.push('/p:WebPublishMethod=FileSystem');
+                overrideParams.push('/p:DeleteExistingFiles=false');
+                overrideParams.push('/p:publishUrl=' + config.websiteRoot);
+                overrideParams.push('/m');
+                overrideParams.push('/restore');
+                msbuild.config('overrideParams', overrideParams);
+                msbuild.config('version', config.buildToolsVersion);
+                msbuild.build();
+            });
+
+    You can alternatively use Sitecore TDS or Helix Publishing Pipeline
+    for local deployments, as seen in the new Sitecore Helix Examples.
+
+
+Are there other tooling improvements which can help?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If your primary concern is the overhead of creating projects which follow
+the Sitecore Helix conventions (naming, filesystem structure, etc.), then
+you might consider some of the community tools which make this easier by
+providing project "templates" which save you some clicks. Take care when
```

Comment:
I was hesitant around SDK-style projects since they are not officially supported for ASP.NET on full framework, but lmk what you two think of this language:

https://github.com/Sitecore/Helix.Docs/commit/4b204301f419398a48f6c31aca15bf022181f31b